oracle/blockeligibilityvalidator.go

```diff
@@ -0,0 +1,70 @@
+package oracle
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+	"github.com/spacemeshos/sha256-simd"
+)
+
+type VRFValidationFunction func(message, signature, publicKey []byte) error
+
+type MinerBlockEligibilityValidator struct {
+	committeeSize  int32
+	layersPerEpoch uint16
+	activationDb   ActivationDb
+	beaconProvider *EpochBeaconProvider
+	validateVRF    VRFValidationFunction
+}
+
+func (v MinerBlockEligibilityValidator) BlockEligible(layerID mesh.LayerID, nodeID mesh.NodeId,
+	proof mesh.BlockEligibilityProof, atxID mesh.AtxId) (bool, error) {
+
+	epochNumber := uint64(layerID) / uint64(v.layersPerEpoch)
+
+	atx, err := v.activationDb.GetAtx(atxID)
+	if err != nil {
+		log.Error("getting ATX failed: %v", err)
+		return false, err
+	}
+
+	if err := atx.Validate(); err != nil {
+		log.Error("ATX is invalid: %v", err)
+		return false, err
+	}
+	if atxEpochNumber := uint64(atx.LayerIndex) / uint64(v.layersPerEpoch); epochNumber != atxEpochNumber {
```

Comment:
extract this to a function, converting from layerId and epoch number is something we do quite often